launcher/sdw-notify.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+"""
+Displays a warning to the user if the workstation has been running continuously
+for too long without checking for security updates. Writes output to a logfile,
+not stdout. All settings are in Notify utility module.
+"""
+
+import sys
+
+from sdw_notify import Notify
+from sdw_util import Util
+from PyQt4 import QtGui
+from PyQt4.QtGui import QMessageBox
+
+# Log file name, base directories defined in Util.py
+LOG_FILE = "sdw-notify.log"
+
+
+def main():
+    """
+    Show security warning, if and only if a warning is not already displayed,
+    the preflight updater is running, and certain checks suggest that the
+    system has not been updated for a specified period
+    """
+
+    Util.configure_logging(LOG_FILE)
+    if(Notify.can_obtain_updater_lock() is False):
+        # Preflight updater is already running. Logged.
+        sys.exit(1)
+
+    # Hold on to lock handle during execution
+    lock_handle = Notify.obtain_notify_lock()
+    if lock_handle is None:
+        # Can't write to lockfile or notifier already running. Logged.
+        sys.exit(1)
+
+    warning_should_be_shown = Notify.is_update_check_necessary()
+    if warning_should_be_shown is None:
+        # Data integrity issue with update timestamp. Logged.
+        sys.exit(1)
+    elif warning_should_be_shown is True:
+        show_update_warning()
+
+
+def show_update_warning():
```

Comment:
Given how straightforward the QT logic is here, I think it's fine to leave as is, but if ever we introduce further complexity/stying here, I would recommend we decouple the main from the QT code, and simply instanciate the QT app object in the main, and handle all other QT tasks in a separate file/class